BioCro/DARPA/upload_params.Rmd

```diff
@@ -15,64 +15,138 @@ library(readxl)
+         access_level = "2" 
+         ) %>%
+  select(citation_author, citation_year, citation_title, 
+         species, cultivar, site, treatment, local_datetime, Date, rep,
+         #Vcmax, leaf_respiration_rate_m2, quantum_efficiency, stomatal_slope.BB, cuticular_cond,
+         variable, mean, 
+         leafT, n, SE, access_level) %>%
+  replace(., is.na(.), "")
 ```
 
-Save cleaned parameter as a .csv file in `upload_params_data` folder in `model-vignettes/BioCro/DARPA`. 
+Add a unique entity comprising of date, treatement, and rep
 ```{r}
-if(!dir.exists("upload_params_data/")){
-  dir.create("upload_params_data")
-}
+upload_parameters$entity <- paste0(upload_parameters$Date, "_", upload_parameters$treatment, "_",
+                                   upload_parameters$rep)
+```
 
+Save cleaned parameter as a .csv file in `upload_params_data` folder in `model-vignettes/BioCro/DARPA`. 
+```{r}
 write.csv(upload_parameters, "upload_params_data/phys_parameters.csv", row.names = FALSE)
 ```
 
-Upload that new parameter data file using the [Bulk Upload](http://welsch.cyverse.org:8000/bety/bulk_upload/start_upload) in Welsch BETYdb. The following additional data, which applies across all the parameters, will have to be entered
 
-- site: Donald Danforth Plant Science Center Growth Chamber
-- species: Setaria viridis
-- access_level: Internal & Collaborators
-- cultivar: ME-034
+Split dataframe, first by new data since last  upload, then by parameter.  
+```{r}
+#create new folder with date
+dname <- gsub('-', '', grep(max(upload_parameters$local_datetime), pattern = "(\\D+)", value = T))
+
+if(!dir.exists(paste0("upload_params_data/", dname))){
+  dir.create(paste0("upload_params_data/", dname))
+}
+newdir <- paste0("upload_params_data/", dname)
+
+#previous maximum date
+prevDate <- max(phys$local_datetime)
+
+#split dataframe for each parameter type and save each dataframe as a csv 
+params <- c("Vcmax", "leaf_respiration_rate_m2", "quantum_efficiency", "theta", "stomatal_slope.BB", "cuticular_cond")
+for(p in params){
+  trait_df <- upload_parameters %>%
+    filter(variable == p,
+           local_datetime > prevDate) %>%
+    spread(variable, mean)
+  
+  # If Vcmax, rename columns n and SE; if cuticular_cond, convert units; if theta, remove leafT
+  if(p == "Vcmax") {
+    trait_df2 <- trait_df %>% dplyr::rename(Sample.size = n, Std.err = SE)} else if(p == "cuticular_cond")
+    {trait_df2 <- trait_df %>% mutate(cuticular_cond = round(cuticular_cond, 0))} else if(p == "theta")
+      {trait_df2 <- trait_df %>% select(-leafT)} else
+        {trait_df2 <- trait_df}
+  
+  write.csv(trait_df2, paste0(newdir, "/phys_parameters_", p, ".csv"), row.names = FALSE)
+}
+```
+
+Upload that new parameter data file using the API. NOTE: Must update to latest folder (date), which on the R side is the variable 'dname'. 
+```{bash}
+#change to directory of latest date
+cd /home/jessicaguo/model-vignettes/BioCro/DARPA/upload_params_data/20200905
```

Comment:
Does this not work? 
```suggestion
cd ~/model-vignettes/BioCro/DARPA/upload_params_data/20200905
```

Reply:
This can be left for a future PR, but you should be able to generate that folder name using grep regex for folder names with numbers only, sort and choose only the most recent one, then generate the URL using that output. This would be a good exercise in shell commands.

Reply:
it does, thanks!